source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+cy.route2('**/users/*/comments')
+
+// https://localhost:7777/users/123/comments     <-- matches
+// https://localhost:7777/users/123/comments/465 <-- does not match
+```
+
+### Use glob to match all segments
+
+```javascript
+cy.route2('**/posts/**')
+
+// https://localhost:7777/posts/1            <-- matches
+// https://localhost:7777/posts/foo/bar/baz  <-- matches
+// https://localhost:7777/posts/quuz?a=b&1=2 <-- matches
+```
+
+### Modify request
+
+Before a request gets passed through, we can modify the request before it is sent out through a function. The function exposes a `req` object which contains the request.
+
+You can also (optionally) add a simulated {% urlHash 'delay' Simulate-delay %} or {% urlHash 'throttle' Simulate-throttle %} on the request to be sent out.
+
+```javascript
+cy.route2('/users', (req) => {
+  // modify the headers or body
+  req.headers = {...req.headers, accept: 'application/json'}
+  req.body = {...req.body, foo: 'bar'}
+})
+```
+
+### Modify response
+
+After a request that gets passed through, we can modify the response from the server before it is returned through a function. The function exposes a `req` object which contains the request. The `req.send()` function exposes a `res` object which contains the response.
+
+```javascript
+cy.route2('/users', (req) => {
+  // passing a function to req.send causes the request to pass through
+  // and allows the response from the origin server to be modified
+  req.send((res) => {
+    res.status = 200
+    res.headers['x-new-headers'] = 'from-server'
+
+    // dynamically alter body
+    res.body = res.body.replace('window.top', 'window.self')
+  })
+})
+```
+
+## With Stubbing
+
+If you pass a `response` to `cy.route()`, Cypress will stub the response in the request.
+
+### `url` as a string
+
+When passing a `string` as the `url`, the URL must match *exactly* what you've written. You'll want to use the decoded string and not include any hash encoding (ie. use `@` instead of `%40`).
+
+```javascript
+cy.route2('https://localhost:7777/surveys/customer?email=john@doe.com', [
+  {
+    id: 1,
+    name: 'john'
+  }
+])
+```
+
+### `url` as a RegExp
+
+When passing a RegExp as the `url`, the url will be tested against the regular expression and will apply if it passes.
+
+```javascript
+cy.route2(/users\/\d+/, { id: 1, name: 'Phoebe' })
+```
+
+```javascript
+// Application Code
+$.get('https://localhost:7777/users/1337', (data) => {
+  console.log(data) // => {id: 1, name: "Phoebe"}
+})
+```
+
+### Response functions
+
+You can also use a function as a response which enables you to add logic and modify properties of the response. The function exposes a `req` object which contains the request as well as a way to `send` with a response.
+
+```javascript
+cy.route2('/login', (req) => {
+  req.send('Success')
+})
+```
+
+```javascript
+// reply with body and headers
+cy.route2('/login', (req) => {
+  req.send('Success', { 'x-new-headers': 'from-server' })
+})
```

Comment:
strange example, this would be more straightforward:

```suggestion
// reply with body and headers
cy.route2('/login', {
  body: 'Success', 
  headers: { 'x-new-headers': 'from-server' }
})
```

this is important b/c when the response can be supplied statically, it is no longer necessary to make serializations and round-trips to the browser, so it significantly improves performance if the user short-circuits the response by supplyin a `StaticResponse` as early as possible. the examples should reflect this - we should refrain from using dynamic request/response interception unless we actually need the dynamic features for an example